lib/util/include/hse_util/event_counter.h

```diff
@@ -66,12 +69,13 @@ extern struct dt_element_ops event_counter_ops;
 
 /* clang-format on */
 
-/* NON-API Functions */
+#if HSE_MOCKING
 void
 ev_get_timestamp(atomic_ulong *timestamp);
 
 size_t
 snprintf_timestamp(char *buf, size_t buf_sz, atomic_ulong *timestamp);
+#endif
```

Comment:
Can you find a way to not use `#ifdef HSE_MOCKING` in the library code?

Reply:
I can just copy the decl to the files where I need it if you want?

`MFT_STATIC` is defined based on `HSE_MOCKING`, so I thought it would be a fine fit.

I don't really have any good ideas.

Reply:
Isn't our library code loaded to the gills with mocking stuff?  defines, comments, includes, ...  it's probably an indication that we're doin' it wrong!  :-)

Reply:
Yea, the organization is not currently ideal, like that alloc.h include in ikvdb.h, ugh

Reply:
I don't want to get into the habit of having source code conditionally compiled if mocking is enabled.  The code @beckerg refers to is annotations which are much different IMO that conditionally compiled source.

Reply:
> I can just copy the decl to the files where I need it if you want?

So you want this?

Reply:
So what's the solution?  I think sometimes it's necessary, or maybe we should just not write unit tests that need extra support from the lib?  I would lean toward that, because at that point it does seem you've gone beyond what it means to be a unit test.

At any rate, I have moved a few such functions into the #if HSE_MOCKING block at the end of the file.  I would rather it be there than scattered throughout the file.

Reply:
> not write unit tests that need extra support from the lib?   I would lean toward that, because at that point it does seem you've gone beyond what it means to be a unit test.

I agree.  Sometimes code that is hard to test can be rewritten a bit to make it more testable.  That's a better option than using ifdefs to change the behavior of the code under test.